paddle/py_paddle/dataprovider_converter.py

```diff
@@ -109,7 +109,10 @@ def pre_scan(self, dat):
             if len(self.__shape__) > 3:
                 raise ValueError(
                     "The dimension of input cannot be greater than 3.")
-            self.__dim__ = reduce(lambda x, y: x * y, self.__shape__)
+            if len(self.__shape__) == 0:
+                self.__dim__ = 1
```

Comment:
After discuss with @qingqing01 , we think if user configure to use a dence_vector, then the data type should be a vector and shape should not be zero